daemon/daemon.go

```diff
@@ -250,15 +250,9 @@ func (d *Daemon) Dying() <-chan struct{} {
 	return d.tomb.Dying()
 }
 
-func (d *Daemon) auther(r *http.Request) (store.Authenticator, error) {
+func (d *Daemon) auther(user *auth.UserState) (store.Authenticator, error) {
```

Comment:
I wasn't suggesting keeping DeviceState in UserState. That'd be awkward. The suggestion was to have an authenticator that can do both.

That said, yes, the changes in #1521 change the picture somewhat, and to a good direction I hope. There are more changes coming that should simplify your life.

I'm getting increasingly concerned about the pace of this development, and how hard it has been to keep in touch (which is certainly the reason for the slow pace). We need to discuss this at the sprint and come up with a plan that is more likely to work.